### PR TITLE
Add PostgreSQL composition

### PR DESCRIPTION
**Pull request by ZuGao**
* Add composition for exoscale postgres database.
* Added commodore config parameters for the package
* Added golden test

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: packages/composition/dbaas/exoscale/postgres.yml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                # Set maintenance time of day
+                - type: CombineFromComposite
+                  combine:
+                    variables:
+                      - fromFieldPath: spec.parameters.maintenance.hour
+                      - fromFieldPath: spec.parameters.maintenance.minute
+                    strategy: string
+                    string:
+                      fmt: "%s:%s:00"
+                  toFieldPath: spec.forProvider.maintenance.timeOfDay
```
Comment (ccremer) [inline packages/composition/dbaas/exoscale/postgres.yml:112]: Why not keep the timeOfDay in the same format in the XRD? (hh:mm:ss). It doesn't make sense to translate [hour,minute] (XRD) -> [timeOfDay] (CRD) -> [hour,minute] (Exoscale API) . Specifying a time of day is more user-friendly than dedicated parameters for hour + minute.

File: packages/composition/dbaas/exoscale/postgres.yml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                # Set backup time of day
+                - type: CombineFromComposite
+                  combine:
+                    variables:
+                      - fromFieldPath: spec.parameters.backup.hour
+                      - fromFieldPath: spec.parameters.backup.minute
+                    strategy: string
+                    string:
+                      fmt: "%s:%s:00"
+                  toFieldPath: spec.forProvider.backup.timeOfDay
```
Comment (ccremer) [inline packages/composition/dbaas/exoscale/postgres.yml:122]: Same here

File: docs/modules/ROOT/pages/references/composition/dbaas/exoscale/common.adoc [hunk 3]
```diff
@@ -0,0 +1,28 @@
+= pkg.appcat.composition.dbaas.exoscale
+
+The parent key for all of the following parameters is `pkg.appcat.composition.dbaas.exoscale`.
+
+Parameters under this parent key apply to all compositions of `pkg.appcat.composition.dbaas.exoscale.*` services.
+
```
Comment (ccremer) [inline docs/modules/ROOT/pages/references/composition/dbaas/exoscale/common.adoc:6]: I'm not sure if the service is really called `dbaas`. Do you have a link or some other source?

File: packages/composition/dbaas/exoscale/postgres.yml (hunk 1 above)
Comment (ZuGao) [reply packages/composition/dbaas/exoscale/postgres.yml:112]: It was proposed like that in the AC. @Kidswiss do you see any issues if we keep the time format the same as from the managed resource?

File: docs/modules/ROOT/pages/references/composition/dbaas/exoscale/common.adoc (hunk 3 above)
Comment (ZuGao) [reply docs/modules/ROOT/pages/references/composition/dbaas/exoscale/common.adoc:6]: it's DBaaS Services on Exoscale.

Comment (ccremer) [reply docs/modules/ROOT/pages/references/composition/dbaas/exoscale/common.adoc:6]: Yes I know Exoscale calls it DBaaS. But does _VSHN_ call it that too?
Compare with S3, VSHN calls the AppCat service "Object storage" : https://products.docs.vshn.ch/products/appcat/objectstorage.html

Comment (ZuGao) [reply docs/modules/ROOT/pages/references/composition/dbaas/exoscale/common.adoc:6]: @Kidswiss can you help here with the naming? Was it decided how to name the exoscale DBaaS Services in AppCat?

File: packages/composition/dbaas/exoscale/postgres.yml [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                - type: PatchSet
+                  patchSetName: annotations
+                - type: PatchSet
+                  patchSetName: labels
+                - type: FromCompositeFieldPath
+                  fromFieldPath: metadata.labels[crossplane.io/composite]
+                  toFieldPath: metadata.name
+                - type: ToCompositeFieldPath
+                  fromFieldPath: status.conditions
+                  toFieldPath: status.accessUserConditions
```
[the remark this answers is not in the document]
Comment (ZuGao) [reply packages/composition/dbaas/exoscale/postgres.yml:79]: Ups... leftover thanks for spotting it

File: packages/composition/dbaas/exoscale/postgres.yml (hunk 1 above)
Comment (Kidswiss) [reply packages/composition/dbaas/exoscale/postgres.yml:109]: If it's easier go for @ccremer's suggestion.

File: docs/modules/ROOT/pages/references/composition/dbaas/exoscale/common.adoc (hunk 3 above)
Comment (Kidswiss) [reply docs/modules/ROOT/pages/references/composition/dbaas/exoscale/common.adoc:6]: Yes, we'll use DBaaS for this.

Comment (ZuGao) [reply docs/modules/ROOT/pages/references/composition/dbaas/exoscale/common.adoc:6]: I guess then it's ok as is?

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**